tasks/configure.yml

```diff
@@ -18,11 +18,12 @@
     group: '{{ headscale_user_gid }}'
     mode: '0600'
   notify: restart headscale
-
+  
 - name: Copy ACL policies file
+  when: headscale_acl_file
   ansible.builtin.copy:
-    content: '{{ headscale_acl | to_yaml }}'
-    dest: '{{ headscale_config_dir }}/acl.yaml'
+    src:  '{{ headscale_acl_file }}'
+    dest: '{{ headscale_config_dir }}/acl.json'
```

Comment:
Let's call it `acl.hujson` to avoid confusion since technically it's not a normal JSON.